File: answer_1.ts
const checker = require('./checker')

// Modify this block
const pairer = (input: string) => {

    return ['ab', 'ac', 'ba', 'bc', 'ca', 'cb']
}
// Modify this block

const input1 = "abc"
let output1: string[] = pairer(input1)
let answer1: string[] = ['ab', 'ac', 'ba', 'bc', 'ca', 'cb']
checker.checkAnswer1(output1, answer1)
console.log("PASSED 1")


const input2 = "bba"
let output2: string[] = pairer(input2)
let answer2: string[] = ['bb', 'ba', 'ab']
checker.checkAnswer1(output2, answer2)
console.log("PASSED 2")

const input3 = "bbbcd"
let output3: string[] = pairer(input3)
let answer3: string[] = ['bb', 'bc', 'bd', 'cb', 'cd', 'db', 'dc']
checker.checkAnswer1(output3, answer3)
console.log("PASSED 3")

File: answer_1.py
# modify your code here
def answer(string):
    result = ["ab", "ac", "ba", "bc", "ca", "cb"]
    print(result)
    return result


if __name__ == "__main__":
    result = answer("abc")
    correct_answer = ["ab", "ac", "ba", "bc", "ca", "cb"]
    assert len(result) == len(correct_answer)
    assert all([each in correct_answer for each in result])
    assert all([each in result for each in correct_answer])
    print("PASSED 1")

    result = answer("bba")
    correct_answer = ["bb", "ba", "ab"]
    assert len(result) == len(correct_answer)
    assert all([each in correct_answer for each in result])
    assert all([each in result for each in correct_answer])
    print("PASSED 2")

    result = answer("bbbcd")
    correct_answer = ["bb", "bc", "bd", "cb", "cd", "db", "dc"]
    assert len(result) == len(correct_answer)
    assert all([each in correct_answer for each in result])
    assert all([each in result for each in correct_answer])
    print("PASSED 3")
